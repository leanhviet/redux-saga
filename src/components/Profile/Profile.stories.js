// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Profile from './index'

storiesOf('Profile', module)
  .add('Default', () => <Profile logout />)
  .add('Contain content', () => (
    <Profile name="Le Van A" subName="Agility IO" logout />
  ))
