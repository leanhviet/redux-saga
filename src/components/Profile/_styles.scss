.profile {
  display: flex;

  &__img {
    width: 40px;
    height: 40px;
    margin-right: 20px;
  }

  &__info {
    width: 90%;

    &__username,
    &__logout {
      font-weight: bold;
    }

    &__username {
      margin: 0;
    }

    &__logout {
      color: $red;
    }
  }
}
