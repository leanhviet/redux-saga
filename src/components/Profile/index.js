// @flow
import React from 'react'
import ImageTheme from '../../themes/images'

type Props = {
  src: string,
  name: string,
  subName: string
}

const Profile = (props: Props) => {
  const { src, name, subName } = props
  const defaultSrc = ev => (ev.target.src = ImageTheme.profile)

  return (
    <div className="profile">
      <img
        className="img-circle profile__img"
        src={src}
        onError={defaultSrc}
        alt="profile"
      />
      <div className="text-left profile__info">
        <p className="profile__info__username">{name}</p>
        <p className="profile__info__subs">{subName}</p>
      </div>
    </div>
  )
}

Profile.defaultProps = {
  src: '',
  name: 'Le Anh Viet',
  subName: '9 Subscribers',
  onLogout: () => {}
}

export default Profile
