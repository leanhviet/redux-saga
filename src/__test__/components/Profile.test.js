// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Themes
import ImageTheme from '../../themes/images'

// Component
import Profile from '../../components/Profile'

const mockProps = {
  src: ImageTheme.logo,
  name: 'Le Van BB',
  subName: 'Agility IO',
  logout: true
}

const wrapper = (props = {}) => {
  const component = shallow(<Profile {...props} />)

  return component
}

describe('Profile component', () => {
  const wrapperWithoutProps = wrapper()

  // Snapshot
  test('should render correctly Profile component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
  })

  // Testing props
  describe('Handle with props', () => {
    test('should is rendered default props', () => {
      expect(
        wrapperWithoutProps.find('.profile__info__username').props().children
      ).toMatch('Le Anh Viet')
      expect(
        wrapperWithoutProps.find('.profile__info__subs').props().children
      ).toMatch('9 Subscribers')
    })

    test('should is rendered with a custom Props', () => {
      const wrapperWithProps = wrapper(mockProps)

      expect(
        wrapperWithProps.find('.profile__info__username').props().children
      ).toMatch(mockProps.name)
      expect(
        wrapperWithProps.find('.profile__info__subs').props().children
      ).toMatch(mockProps.subName)
    })
  })
})
