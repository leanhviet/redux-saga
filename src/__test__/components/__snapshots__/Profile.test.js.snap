// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render correctly Profile component default props 1`] = `
<div
  className="profile"
>
  <img
    alt="profile"
    className="img-circle profile__img"
    onError={[Function]}
    src=""
  />
  <div
    className="text-left profile__info"
  >
    <p
      className="profile__info__username"
    >
      Le Anh Viet
    </p>
    <p
      className="profile__info__subs"
    >
      9 Subscribers
    </p>
  </div>
</div>
`;
